currency/code.go

```diff
@@ -187,9 +187,9 @@ func (b *BaseCodes) Register(c string, newRole Role) Code {
 	}
 
 	var format bool
-	// Digits fool upper and lower casing. So find first letter and check case.
 	for x := range c {
 		if unicode.IsLetter(rune(c[x])) {
+			// Digits fool upper and lower casing. So find first letter and check case.
 			format = unicode.IsUpper(rune(c[x]))
```

Comment:
Interesting that this would have a strange outcome on a mixed case code, like sETH.
Feeling like this could be simplifier to use strings.ContainsFunc(... unicode.IsLetter() && unicode.IsUpper() )
Then we'd consider it uppercase if anything in it is uppercase.

Reply:
```suggestion
	format := strings.ContainsFunc(c, func(r rune){ return unicode.IsLetter(r) && unicode.IsUpper(r) })
```

Reply:
🔥 I like this. 👍